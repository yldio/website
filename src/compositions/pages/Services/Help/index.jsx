/* eslint-disable global-require */

import React from 'react';
import styled from 'styled-components';
import remcalc from 'remcalc';
import is, { isNot } from 'styled-is';

import Container from 'components/Container';
import { pink } from 'styles/colours';
import Card from 'components/Card';
import { H2 } from 'components/Typography';

import { Section } from '../styled';

const Bar = styled.div`
  background: #5e5cb2;
  width: ${remcalc(100)};
  height: ${remcalc(4)};
  margin-top: ${remcalc(15)};
  margin-bottom: ${remcalc(70)};
`;

const Wrapper = styled(Container)`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  ${isNot('noMargin')`
    margin-top: ${remcalc(-140)};
  `}
  margin-bottom: ${remcalc(70)};

  ${is('noMargin')`
    padding: 0;
  `}
`;

const CardsWrapper = styled.div`
  background: #190533;
  padding-top: ${remcalc(70)};
  padding-bottom: ${remcalc(70)};
  margin-top: ${remcalc(140)};
`;

const HelpSection = styled(Section)`
  padding: 0;
  padding-top: ${remcalc(140)};
  margin-top: 0px;
`;

const Item = styled.li`
  &::before {
    content: 'â€¢';
    margin-right: ${remcalc(20)};
    color: ${pink};
  }
  &:not(:last-child) {
    margin-bottom: ${remcalc(20)};
  }

  color: white;
`;

const Help = () => (
  <HelpSection data-selector="services:tech">
    <Container>
      <H2 white slim>
        We bring you the latest techniques<br /> to help you meet your
        challenges
        <Bar />
      </H2>
    </Container>
    <CardsWrapper>
      <Wrapper>
        <Card
          title="YLD DELIVERS"
          points={['Lean Culture', 'Micro-services']}
        />
        <Card
          title="Leveraging"
          points={['Vertical teams', 'Open source software']}
        />
        <Card
          title="Enabling"
          points={['Experimentation', '	Continuous delivery with zero downtime']}
        />
      </Wrapper>
      <Container>
        <H2 white slim>
          To bring you business value
          <Bar />
        </H2>
        <Wrapper noMargin>
          <ul>
            <Item>Utilise disruptive technology</Item>
            <Item>Stay lean and compete</Item>
          </ul>
          <ul>
            <Item>Transform</Item>
            <Item>Innovate</Item>
          </ul>
          <ul>
            <Item>Speed of delivery and efficiency</Item>
            <Item>Bridge the skills gap</Item>
          </ul>
          <ul>
            <Item>Mitigate risk</Item>
            <Item>Lower cost</Item>
          </ul>
        </Wrapper>
      </Container>
    </CardsWrapper>
  </HelpSection>
);

export default Help;
