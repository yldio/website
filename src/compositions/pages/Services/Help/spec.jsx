import React from 'react';
import { shallow } from 'enzyme';
import Help from './index';

test('compositions/pages/Services/Help snapshot', () => {
  const wrapper = shallow(<Help />);

  expect(wrapper).toMatchSnapshot();
});
