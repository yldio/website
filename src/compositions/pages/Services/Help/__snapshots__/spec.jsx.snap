// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compositions/pages/Services/Help snapshot 1`] = `
<Help__HelpSection
  data-selector="services:tech"
>
  <Container>
    <Typography__H2
      slim={true}
      white={true}
    >
      We bring you the latest techniques
      <br />
       to help you meet your challenges
      <Help__Bar />
    </Typography__H2>
  </Container>
  <Help__CardsWrapper>
    <Help__Wrapper>
      <Card
        points={
          Array [
            "Lean Culture",
            "Micro-services",
          ]
        }
        title="YLD DELIVERS"
      />
      <Card
        points={
          Array [
            "Vertical teams",
            "Open source software",
          ]
        }
        title="Leveraging"
      />
      <Card
        points={
          Array [
            "Experimentation",
            "	Continuous delivery with zero downtime",
          ]
        }
        title="Enabling"
      />
    </Help__Wrapper>
    <Container>
      <Typography__H2
        slim={true}
        white={true}
      >
        To bring you business value
        <Help__Bar />
      </Typography__H2>
      <Help__Wrapper
        noMargin={true}
      >
        <ul>
          <Help__Item>
            Utilise disruptive technology
          </Help__Item>
          <Help__Item>
            Stay lean and compete
          </Help__Item>
        </ul>
        <ul>
          <Help__Item>
            Transform
          </Help__Item>
          <Help__Item>
            Innovate
          </Help__Item>
        </ul>
        <ul>
          <Help__Item>
            Speed of delivery and efficiency
          </Help__Item>
          <Help__Item>
            Bridge the skills gap
          </Help__Item>
        </ul>
        <ul>
          <Help__Item>
            Mitigate risk
          </Help__Item>
          <Help__Item>
            Lower cost
          </Help__Item>
        </ul>
      </Help__Wrapper>
    </Container>
  </Help__CardsWrapper>
</Help__HelpSection>
`;
